hs_core/tests/api/native/test_irods_user_zone_federation.py

```diff
@@ -109,7 +109,20 @@ def test_resource_operations_in_user_zone(self):
         self.assertEqual(res.resource_federation_path, fed_path)
         # test original file in user test zone is removed after resource creation
         # since True is used for move when creating the resource
-        self.assertFalse(self.irods_storage.exists(user_path + self.file_to_be_deleted))
+        file_path_name = user_path + self.file_to_be_deleted
+        self.assertFalse(self.irods_storage.exists(file_path_name))
+
+        # test django_irods CopyFiles() with an iRODS resource name being passed in
+        # as input parameter to verify the file gets copied to the pass-in iRODS resource
+        istorage = res.get_irods_storage()
+        src_path = os.path.join(res.root_path, 'data', 'contents', self.file_to_be_deleted)
+        dest_path = file_path_name
+        istorage.copyFiles(src_path, dest_path, settings.HS_IRODS_LOCAL_ZONE_DEF_RES)
+        # assert file did get copied over
+        self.assertTrue(self.irods_storage.exists(file_path_name))
+        stdout = self.irods_storage.session.run("ils", None, "-l", file_path_name)[0].split()
+        # assert copied file gets written to the iRODS resource being passed into copyFiles() call
+        self.assertEqual(stdout[2], settings.HS_IRODS_LOCAL_ZONE_DEF_RES)

```

Comment:
Is there a reason this can't be broken out into it's own test?  It's generally best practice to keep unit tests small and only test on functionality at a time.

Reply:
@sblack-usu I'd prefer to leave this test case bundled together with the rest of test cases in this ```test_resource_operations_in_user_zone()``` function for logical coherence, since the preceding test cases for testing file "move" or "copy" operations all call the same copyFiles() routine in django_irods, and the additional iRODS resource pass-in option added to copyFiles() routine natually fit in there in my opinion. In fact, the whole file operation test cases enclosed in this test routine are all logically connected, so I think there is merit to bundle them together in this case.